lib/hexpm/repository/packages.ex

```diff
@@ -71,4 +71,16 @@ defmodule Hexpm.Repository.Packages do
     |> Repo.all
     |> Enum.into(%{})
   end
+
+  def dependent_package_ids(name) do
+    from(rq in Requirement,
+      join: p in Package,
+      join: rl in Release,
+      select: rl.package_id,
+      distinct: rl.package_id,
+      where: rq.dependency_id == p.id,
+      where: rl.id == rq.release_id,
```

Comment:
You can do `join: rl in assoc(rq, :release)` instead of using the `:where`, it expands to `join: rl in Release, on: rl.id == rq.release_id`.